builtin/providers/kubernetes/schema_pod_spec.go

```diff
@@ -0,0 +1,139 @@
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Description: "NodeSelector is a selector which must be true for the pod to fit on a node. Selector which must match a node's labels for the pod to be scheduled on that node. More info: http://kubernetes.io/docs/user-guide/node-selection.",
+		},
+		"restart_policy": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      "Always",
+			Description:  "Restart policy for all containers within the pod. One of Always, OnFailure, Never. More info: http://kubernetes.io/docs/user-guide/pod-states#restartpolicy.",
+			ValidateFunc: validateRestartPolicy,
+		},
+		"service_account_name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "ServiceAccountName is the name of the ServiceAccount to use to run this pod. More info: http://releases.k8s.io/HEAD/docs/design/service_accounts.md.",
+		},
+		"subdomain": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `If specified, the fully qualified Pod hostname will be "...svc.". If not specified, the pod will not have a domainname at all..`,
+		},
+		"termination_grace_period_seconds": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Default:      30,
+			Description:  "Optional duration in seconds the pod needs to terminate gracefully. May be decreased in delete request. Value must be non-negative integer. The value zero indicates delete immediately. If this value is nil, the default grace period will be used instead. The grace period is the duration in seconds after the processes running in the pod are sent a termination signal and the time when the processes are forcibly halted with a kill signal. Set this value longer than the expected cleanup time for your process.",
+			ValidateFunc: validateTerminationGracePeriodSeconds,
+		},
+		"image_pull_secrets": {
+			Type:        schema.TypeList,
+			Description: "Optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec. If specified, these secrets will be passed to individual puller implementations for them to use. For example, in the case of docker, only DockerConfig type secrets are honored. More info: http://kubernetes.io/docs/user-guide/images#specifying-imagepullsecrets-on-a-pod",
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:        schema.TypeString,
+						Description: "Name of the referent. More info: http://kubernetes.io/docs/user-guide/identifiers#names",
+						Optional:    true,
+					},
+				},
+			},
+		},
+		"volumes": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "List of volumes that can be mounted by containers belonging to the pod. More info: http://kubernetes.io/docs/user-guide/volumes",
+			Elem:        volumeSchema(),
+		},
+		"containers": {
+			Type:        schema.TypeList,
+			Optional:    true,
```

Comment:
Given that deployment or replication controllers etc may use pod's schema and they would allow the containers to be updated, sort of, basically they would destroy and create a new set of containers internally. So I thought it would be better to leave it as it is. As providing ForceNew and using the same schema in deployment would lead to deployment destruction. I am not very sure how you would typically handle this one. Perhaps by not making the deployment depend on the pod schema?. This would lead to lots of code duplication though.